articles/building-a-simple-cryptocurrency-blockchain/index.md

```diff
@@ -0,0 +1,187 @@
+        }
+        initGenesisBlock(){
+            return new BlockCrypto(0, "06/04/2021", "Initial Block in the Chain", "0");
+        }
+        latestBlock(){
+            return this.block1chain[this.block1chain.length - 1];
+        }
+        addNewBlock(newBlock){
+            newBlock.nextHash = this.latestBlock().hash;
+            newBlock.hash = newBlock.computeHash();        
+            this.block1chain.push(newBlock);
+        }
+    }
+```js
+
+
+Let's explain the above class 
+
+As usual. we have our constructor which instantiates the block1chain but this time we passed to it the `initGenesisBlock()` method which initializes the block in the chain. This property refers to an array of blocks in our case.
+
+- `initGenesisBlock()`: This is the first block created in the peer-to-peer network and has not been linked to any other. To our knowledge of indexing it's at index 0, Notice we created it using the previously created 'BlockCrypto' class and passed all the parameters.
+
+- `latestBlock` : As named, we can easily tell it's finding the latest block in the chain. As explained earlier it helps to ensure the hash of the current block and map it to the hash of the previous block to ensure chain integrity.
+
+- `addNewBlock`: This method I used here to add a new block to the chain. Matching the previous hash block and current hash block in our case is to ensure we are not tampering with the chain.
+
+
+Now that our blockchain is ready to work. We are missing something which is a core blockchain principle, the blockchain _integrity_. Let's see how to verify it and test our app.
+
+## Verifying Blockchain Integrity 
+
+The main characteristic of blockchain is that once a block has been added to the network, it can't be changed without invalidating the entire blockchain integrity.
+
+To perform this, we use digital security or cryptographic hash which ensures securing and validating of blockchain by producing a new hash every time any change is made in the block thus invalidating the blockchain.
+
+We'll loop over the entire blockchain to check whether any hash has been tampered with, taking into account the exception of the first block which is hardcoded.
+
+Besides, this method will verify if the hashes of each two consecutive blocks are pointing to one another. If the integrity of the blockchain has been compromised, it returns false; otherwise, in case no anomalies are encountered, it returns true.
+ We'll create this method inside inside `Blockchain` class
+```Node
+    checkValidity(){
+            for(let i = 1; i < this.block1chain.length; i++){
+                const currentBlock = this.block1chain[i];
+                const precedingBlock= this.blockchain[i-1];
+
+            if(currentBlock.hash !== currentBlock.computeHash()){
+                return false;
+            }
+            if(currentBlock.nextgHash !== nextBlock.hash)
+                return false;
+            }
+            return true;
+        }
+```js
```

Comment:
I do not find an explanation for this. So, add comments

Reply:
Adding comments asap!